detectron2/structures/keypoints.py

```diff
@@ -159,6 +158,12 @@ def heatmaps_to_keypoints(maps: torch.Tensor, rois: torch.Tensor) -> torch.Tenso
     we maintain consistency with :meth:`Keypoints.to_heatmap` by using the conversion from
     Heckbert 1990: c = d + 0.5, where d is a discrete coordinate and c is a continuous coordinate.
     """
+    # functions decorated with @torch.no_grad() can not be exported to torchscript when
+    # other functions are called inside them.
```

Comment:
Nothing needed to be done for this PR, but just curious:
The decorator use was not supported according to this PR: https://github.com/pytorch/pytorch/pull/41371
Did you mean that decorator it's actually scriptable if the function does not call other functions inside? If that's true I wonder whether `no_grad` is effective or is simply not parsed by jit -- the latter case should be considered a bug.

Reply:
Yes, if no other functions are called inside, the target function can be exported successfully. (Not sure if this applies to all functions in that situation, I just test several simple functions).

For now I have no ideas about this phenomenon, I can elaborate this in https://github.com/pytorch/pytorch/issues/44768

Reply:
This seems to be a bug of JIT, I will report it in that issue.